### PR TITLE
Set default pytest timeout of 240s

### LINKED ISSUE #7217: Add default timeout for tests
**Describe the requested improvement**
It would be great if pytest would time out instead of a github action runner timing out. This way, we would know what test is hanging. It would be fairly easy to add something along the lines of

```
if item.get_marker('timeout') is None:
            item.add_marker(pytest.mark.timeout(60))
```
in conftest.

**Expected behaviour**
We should know what test is hanging.

### PR DESCRIPTION
**Pull request by jonathan-eq**
**Issue**
Resolves #7217 

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

**Comment by berland** [comment]
I would claim this is not setting a "default" timeout since any test not having the marker will not time out?

Since this requires an explicit marker on every test, what about just having a custom timeout for each test that could time out (then I am thinking about many tests that I know could have only say 5 seconds timeout)?

**Comment by jonathan-eq** [comment]
> I would claim this is not setting a "default" timeout since any test not having the marker will not time out?
> 
> Since this requires an explicit marker on every test, what about just having a custom timeout for each test that could time out (then I am thinking about many tests that I know could have only say 5 seconds timeout)?

I think this would add the marker to any test not already having the marker.
I will try setting it in pyproject instead, just to make sure.

The idea was to set an upper limit for all tests, and make sure pytest will timeout the test before github actions times out pytest. This way we will know what test is hanging. The timeout period can be anything less than the workflow timeout, but we should be sure the potentially hanging test is actually hanging (so maybe 5-7 minutes?)

File: .github/workflows/test_ert.yml [hunk 1]
```diff
@@ -23,6 +22,7 @@ jobs:
 
     - uses: actions/setup-python@v4
       id: setup_python
+      timeout-minutes: 30
```
Comment (berland) [inline .github/workflows/test_ert.yml:25]: At this level, we can go much lower than 30 minutes :)

**Review by berland — APPROVED** [review]
Lets go!

**Pull request merged**